lib/debug/repl.hpp

```diff
@@ -0,0 +1,36 @@
+#include <iostream>
+#include <string>
+#include <vector>
+#include <unordered_map>
+
+#include <core/logger.hpp>
+#include <io/adc.hpp>
+#include <io/i2c.hpp>
+
+namespace hyped::debug {
+
+struct Command {
+  std::string name;
+  std::string description;
+  std::function<void(void)> callback;
+};
+
+class Repl {
+ public:
+  Repl(hyped::core::ILogger &log);
+  void run();
+
+ private:
+  void printCommands();
+  void addCommand(const Command &cmd);
+  void addQuitCommand();
+  void addHelpCommand();
+  void handleCommand();
+
+  void addAdcCommands(const uint8_t pin);
+
+  std::vector<Command> commands_;
```

Comment:
Unsure what we need this for? If it's to print help you could just use an ordered map and iterate over it? This would have the benefit of putting the commands in alphabetical order.